rskj-core/src/main/java/org/ethereum/config/SystemProperties.java

```diff
@@ -583,42 +587,54 @@ public synchronized String getExternalIp() {
             return externalIp;
         }
 
-        if (configFromFiles.hasPath("peer.discovery.external.ip")) {
-            String externalIpFromConfig = configFromFiles.getString("peer.discovery.external.ip").trim();
+        if (configFromFiles.hasPath(PROPERTY_EXTERNAL_IP)) {
+            String externalIpFromConfig = configFromFiles.getString(PROPERTY_EXTERNAL_IP).trim();
             if (!externalIpFromConfig.isEmpty()){
                 try {
-                    InetAddress addr = InetAddress.getByName(externalIpFromConfig);
-                    externalIp = addr.getHostAddress();
+                    InetAddress address = tryParseIpOrThrow(externalIpFromConfig);
+                    externalIp = address.getHostAddress();
                     logger.info("External address identified {}", externalIp);
                     return externalIp;
                 } catch (IOException e) {
-                    externalIp = null;
-                    logger.warn("Can't resolve external address");
+                } catch (IllegalArgumentException e) {
                 }
+                externalIp = null;
+                logger.warn("Can't resolve external address");
             }
         }
 
         externalIp = getMyPublicIpFromRemoteService();
         return externalIp;
     }
 
-    public String getMyPublicIpFromRemoteService(){
-        logger.info("External IP wasn't set or resolved, using checkip.amazonaws.com to identify it...");
+    private String getMyPublicIpFromRemoteService(){
         try {
+            logger.info("External IP wasn't set or resolved, using checkip.amazonaws.com to identify it...");
+
             try (BufferedReader in = new BufferedReader(new InputStreamReader(new URL("http://checkip.amazonaws.com").openStream()))) {
                 externalIp = in.readLine();
             }
 
             if (externalIp == null || externalIp.trim().isEmpty()) {
+                logger.warn("Unable to retrieve external ip from checkip.amazonaws.com {}.", externalIp);
                 throw new IOException("Invalid address: '" + externalIp + "'");
             }
 
-            tryParseIpOrThrow();
+            tryParseIpOrThrow(externalIp);
             logger.info("External address identified: {}", externalIp);
+            return externalIp;
         } catch (IOException e) {
             logger.error("Can't get external IP. " + e);
-            externalIp = getPeerDiscoveryBindAddress();
+        } catch (IllegalArgumentException e) {
+            logger.error("Can't get external IP. " + e);
         }
+
+        String bindAddress = getBindAddress().toString();
+        if (getBindAddress().isAnyLocalAddress()){
+            throw new RuntimeException("Fallback on bind address for external IP doesn't allow wildcard " + bindAddress);
```

Comment:
I had to read it a few times to understand the error message  "Wildcard bind address can't be used as node's public IP"?

Reply:
getBindAddress returns a InetAddress, I would keep it that way. 

"Wildcard on bind address it's not allowed as fallback for public IP" I would like to make clear that wildcard on bind address can't be used as  public IP because it couldn't be retrieved from config nor amazon, it's not the main source for public IP